tests/subsys/zbus/dyn_channel/src/main.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/zbus/zbus.h>
+#include <zephyr/ztest.h>
+#include <zephyr/ztest_assert.h>
+LOG_MODULE_DECLARE(zbus, CONFIG_ZBUS_LOG_LEVEL);
+
+struct version_msg {
+	uint8_t major;
+	uint8_t minor;
+	uint16_t build;
+};
+
+struct external_data_msg {
+	void *reference;
+	size_t size;
+};
+
+ZBUS_CHAN_DEFINE(dyn_chan_no_subs,	   /* Name */
+		 false,			   /* Read only */
+		 struct external_data_msg, /* Message type */
+
+		 NULL,		       /* Validator */
+		 ZBUS_OBSERVERS_EMPTY, /* observers */
+		 ZBUS_MSG_INIT(0)      /* Initial value {0} */
+);
+
+ZBUS_CHAN_DEFINE(dyn_chan,		   /* Name */
+		 false,			   /* Read only */
+		 struct external_data_msg, /* Message type */
+
+		 NULL,		     /* Validator */
+		 ZBUS_OBSERVERS(s1), /* observers */
+		 ZBUS_MSG_INIT(0)    /* Initial value {0} */
+);
+
+static struct {
+	uint8_t a;
+	uint64_t b;
+} my_random_data_expected = {0};
```

Comment:
Drop the initialization to zero, all global variables are zeroed automatically, saves few bytes of flash.

Reply:
hey this are still here